ClassCastException running Mockito with Quarkus 1.3.0.Final

Using Mockito into @QuarkusTest, like this:
```
    @Test
    public void testHelloEndpoint() {

        HttpServletRequest request = mock(HttpServletRequest.class);
```
I got a ClassCastException:

```
﻿[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.177 s <<< FAILURE! - in net.lucamasini.testcase.ExampleResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.677 s  <<< ERROR!
org.mockito.exceptions.base.MockitoException: 

ClassCastException occurred while creating the mockito mock :
  class to mock : 'javax.servlet.http.HttpServletRequest', loaded by classloader : 'jdk.internal.loader.ClassLoaders$AppClassLoader@6ff3c5b5'
  created class : 'org.mockito.codegen.HttpServletRequest$MockitoMock$188783148', loaded by classloader : 'net.bytebuddy.dynamic.loading.MultipleParentClassLoader@510689af'
  proxy instance class : 'org.mockito.codegen.HttpServletRequest$MockitoMock$188783148', loaded by classloader : 'net.bytebuddy.dynamic.loading.MultipleParentClassLoader@510689af'
  instance creation by : ObjenesisInstantiator

You might experience classloading issues, please ask the mockito mailing-list.

	at net.lucamasini.testcase.ExampleResourceTest.testHelloEndpoint(ExampleResourceTest.java:18)
Caused by: java.lang.ClassCastException: class org.mockito.codegen.HttpServletRequest$MockitoMock$188783148 cannot be cast to class org.mockito.internal.creation.bytebuddy.MockAccess (org.mockito.codegen.HttpServletRequest$MockitoMock$188783148 is in unnamed module of loader net.bytebuddy.dynamic.loading.MultipleParentClassLoader @510689af; org.mockito.internal.creation.bytebuddy.MockAccess is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @65f095f8)
	at net.lucamasini.testcase.ExampleResourceTest.testHelloEndpoint(ExampleResourceTest.java:18)

2020-03-13 16:53:28,727 INFO  [io.quarkus] (main) Quarkus stopped in 0.024s
```
this worked with 1.2.1.Final.

--- a/core/runtime/pom.xml
+++ b/core/runtime/pom.xml
@@ -109,8 +109,17 @@
                         <parentFirstArtifact>org.jboss.logmanager:jboss-logmanager-embedded</parentFirstArtifact>
                         <parentFirstArtifact>org.jboss.logging:jboss-logging</parentFirstArtifact>
                         <parentFirstArtifact>org.ow2.asm:asm</parentFirstArtifact>
+                        <parentFirstArtifact>org.ow2.asm:asm-commons</parentFirstArtifact>
                         <parentFirstArtifact>org.apache.maven:maven-model</parentFirstArtifact>
+                        <!--
+                        This is a temporary fix, we need a better way to be able to
+                        add these exclusions outside the quarkus-extension.properties file.
+                        -->
                         <parentFirstArtifact>org.opentest4j:opentest4j</parentFirstArtifact>
+                        <parentFirstArtifact>org.mockito:mockito-core</parentFirstArtifact>
+                        <parentFirstArtifact>org.objenesis:objenesis</parentFirstArtifact>
+                        <parentFirstArtifact>net.bytebuddy:byte-buddy</parentFirstArtifact>
+                        <parentFirstArtifact>net.bytebuddy:byte-buddy-agent</parentFirstArtifact>
                     </parentFirstArtifacts>
                     <excludedArtifacts>
                         <excludedArtifact>io.smallrye:smallrye-config</excludedArtifact>

--- a/integration-tests/main/pom.xml
+++ b/integration-tests/main/pom.xml
@@ -148,6 +148,11 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
             <groupId>org.webjars</groupId>
             <artifactId>bootstrap</artifactId>
         </dependency>
@@ -156,6 +161,12 @@
             <groupId>org.ops4j.pax.web.samples.web-fragment</groupId>
             <artifactId>fragment</artifactId>
             <version>7.3.3</version>
+            <scope>test</scope>
+        </dependency>
+        <!-- Issue 7854: ensure that mockito works properly -->
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
             <scope>test</scope>
         </dependency>
     </dependencies>

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/MockitoTest.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/MockitoTest.java
@@ -1,0 +1,32 @@
+package io.quarkus.it.main;
+
+import static org.mockito.Mockito.mock;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class MockitoTest {
+
+    @Test
+    public void testMockHttpServletRequest() {
+        assertNoExceptionThrown(HttpServletRequest.class);
+    }
+
+    @Test
+    public void testMockApplicationClass() {
+        assertNoExceptionThrown(Whatever.class);
+    }
+
+    private void assertNoExceptionThrown(Class<?> classToMock) {
+        Assertions.assertThatCode(() -> mock(classToMock)).doesNotThrowAnyException();
+    }
+
+    public static class Whatever {
+
+    }
+}

